src/includes/class-alma-wc-payment-validator.php

```diff
@@ -38,47 +38,71 @@ public static function validate_payment( $payment_id ) {
 		try {
 			$payment = $alma->payments->fetch( $payment_id );
 		} catch ( RequestError $e ) {
-			$logger->error( 'Error while fetching payment with id ' . $payment_id . ': ' . $e->getMessage() );
+			$logger->error( 'Error while fetching payment with id.', [
+                'Method' => __METHOD__ ,
+                'PaymentId' => $payment_id,
+                'ExceptionMessage' => $e->getMessage()
+            ]);
+
 			throw new Alma_WC_Payment_Validation_Error( 'payment_fetch_error' );
 		}
 
 		try {
 			$order = new Alma_WC_Model_Order( $payment->custom_data['order_id'], $payment->custom_data['order_key'] );
 		} catch ( \Exception $e ) {
-			$logger->error( "Error getting order associated to payment '$payment_id': " . $e->getMessage() );
+            $logger->error( 'Error getting order associated to payment.', [
+                'Method' => __METHOD__ ,
+                'PaymentId' => $payment_id,
+                'ExceptionMessage' => $e->getMessage()
+            ]);
+
 			throw new Alma_WC_Payment_Validation_Error( 'order_fetch_error' );
 		}
 
 		if ( $order->get_wc_order()->has_status( apply_filters( 'alma_wc_valid_order_statuses_for_payment_complete', array( 'on-hold', 'pending', 'failed', 'cancelled' ) ) ) ) {
 			if ( $order->get_total() !== $payment->purchase_amount ) {
-				$error = "Order {$order->get_id()} total ({$order->get_total()}) does not match purchase amount of '$payment_id' ($payment->purchase_amount)";
-				$logger->error( $error );
+				$logger->error( 'The total order amount does not match the purchase amount of the payment.', [
+                    'Method' => __METHOD__,
+                    'OrderId' => $order->get_id(),
+                    'OrderAmount' => $order->get_total(),
+                    'PaymentId' => $payment_id,
+                    'PaymentPurchaseAmount' => $payment->purchase_amount
+                ]);
 
 				try {
-					$logger->warning( "Amount mismatch for order {$order->get_id()}" );
 					$alma->payments->flagAsPotentialFraud( $payment_id, Payment::FRAUD_AMOUNT_MISMATCH );
 				} catch ( RequestError $e ) {
-					$logger->warning( $e->getMessage() );
+					$logger->error( 'FlagAsPotentialFraud request error (total amount issue).', [
+                        'Method' => __METHOD__,
+                        'ExceptionMessage'  => $e->getMessage()
+                    ]);
 				}
 
-				throw new Alma_WC_Payment_Validation_Error( $error );
+				throw new Alma_WC_Payment_Validation_Error(sprintf("Order %s total (%s) does not match purchase amount of '%s' (%s)", $order->get_id(), $order->get_total(), $payment_id, $payment->purchase_amount));
```

Comment:
need translation ?

Reply:
Yes, the error is catch and send back to the front